nwb_conversion_tools/utils/nwbephyswriterdatachunkiterator.py

```diff
@@ -1,48 +1,98 @@
-"""Authors: Saksham Sharda."""
-from typing import Tuple, Iterable
+"""Authors: Saksham Sharda, Alessio Buccino."""
 import numpy as np
+from typing import Tuple, Iterable, Union
+from distutils.version import StrictVersion
+
 from .genericdatachunkiterator import GenericDataChunkIterator
 
+from .basenwbephyswriter import BaseNwbEphysWriter
+
 
 class NwbEphysWriterDataChunkIterator(GenericDataChunkIterator):
-    """DataChunkIterator specifically for use on RecordingExtractor objects."""
+    """DataChunkIterator for use on RecordingExtractor (spikeinterface/spikeinterface) objects."""
 
     def __init__(
         self,
-        ephys_writer,
+        ephys_writer: BaseNwbEphysWriter,
         segment_index: int = 0,
         unsigned_coercion: list = None,
-        write_scaled: bool = True,
+        write_scaled: bool = False,
         buffer_gb: float = None,
         buffer_shape: tuple = None,
         chunk_mb: float = None,
         chunk_shape: tuple = None,
+        dtype: Union[str, np.dtype] = None,
```

Comment:
The only thing about adding this new argument is now, I really feel like this function is getting overloaded. Especially considering the interactions between `unsigned_coercion`, `write_scaled`, and `dtype`; these aren't being fully tested in that capacity and at various levels may be redundant.

I understand we will control these ourselves in specified ways at the `add_electrical_series` call, but this Iterator is also intended to be capable of standalone usage.

I'll do a separate PR into this that tries to reduce what's needed to get to the bare minimum of required information.